nixos/modules/services/network-filesystems/moosefs.nix

```diff
@@ -205,42 +213,86 @@ in {
-    ( systemdService "master" {
-      TimeoutStartSec = 1800;
-      TimeoutStopSec = 1800;
-      Restart = "no";
-    } masterCfg );
-
-    systemd.services.mfs-metalogger = lib.mkIf cfg.metalogger.enable
-      ( systemdService "metalogger" { Restart = "on-abnormal"; } metaloggerCfg );
-
-    systemd.services.mfs-chunkserver = lib.mkIf cfg.chunkserver.enable
-      ( systemdService "chunkserver" { Restart = "on-abnormal"; } chunkserverCfg );
-    };
+    environment.systemPackages = lib.optional cfg.client.enable pkgs.moosefs;
+
+    networking.firewall.allowedTCPPorts = lib.mkMerge [
+      (lib.optionals cfg.master.openFirewall [ 9419 9420 9421 ])
+      (lib.optional cfg.chunkserver.openFirewall 9422)
+      (lib.optional (cfg.cgiserver.enable && cfg.cgiserver.openFirewall) cfg.cgiserver.settings.PORT)
+    ];
+
+    systemd.tmpfiles.rules = [
+      # Base directory for MooseFS
+      "d /var/lib/mfs 0755 ${mfsUser} ${mfsUser} -"
+      "d /run/mfs 0755 ${mfsUser} ${mfsUser} -"
+
+      # Master directories
+      (lib.optionalString cfg.master.enable
+        "d ${cfg.master.settings.DATA_PATH} 0700 ${mfsUser} ${mfsUser} -")
+
+      # Metalogger directories
+      (lib.optionalString cfg.metalogger.enable
+        "d ${cfg.metalogger.settings.DATA_PATH} 0700 ${mfsUser} ${mfsUser} -")
+
+      # Chunkserver directories
+      (lib.optionalString cfg.chunkserver.enable
+        "d ${cfg.chunkserver.settings.DATA_PATH} 0700 ${mfsUser} ${mfsUser} -")
+    ] ++ lib.optionals (cfg.chunkserver.enable && cfg.chunkserver.hdds != null)
+        (map (dir: "d ${dir} 0755 ${mfsUser} ${mfsUser} -") cfg.chunkserver.hdds);
+
+    systemd.services = lib.mkMerge [
+      (lib.mkIf cfg.master.enable {
+        mfs-master = systemdService "master" {
+          TimeoutStartSec = 1800;
+          TimeoutStopSec = 1800;
+          Restart = "on-failure";
+          User = mfsUser;
+        } masterCfg;
+      })
+
+      (lib.mkIf cfg.metalogger.enable {
+        mfs-metalogger = systemdService "metalogger" {
+          Restart = "on-abnormal";
+          User = mfsUser;
+        } metaloggerCfg;
+      })
+
+      (lib.mkIf cfg.chunkserver.enable {
+        mfs-chunkserver = systemdService "chunkserver" {
+          Restart = "on-abnormal";
+          User = mfsUser;
+        } chunkserverCfg;
+      })
+
+      (lib.mkIf cfg.cgiserver.enable {
+        mfs-cgiserv = {
+          description = "MooseFS CGI Server";
+          wantedBy = [ "multi-user.target" ];
+          after = [ "mfs-master.service" ];
+
+          serviceConfig = {
+            Type = "simple";
+            ExecStart = "${pkgs.moosefs}/bin/mfscgiserv -D /var/lib/mfs -f start";
+            ExecStop = "${pkgs.moosefs}/bin/mfscgiserv -D /var/lib/mfs stop";
+            Restart = "on-failure";
+            RestartSec = "30s";
+            User = mfsUser;
+            Group = mfsUser;
+            WorkingDirectory = "/var/lib/mfs";
+          };
+        };
+      })
+    ];
+  };
```

Comment:
Maybe add yourself ass an maintainer, by adding something like this, (see below) and to the `maintainers/maintainer-list.nix` file.
```
  meta = {
    maintainers = with lib.maintainers; [ _YOUR_USERNAME  ];
  };
  ```